lib/explorer/data_frame.ex

```diff
@@ -2151,20 +2247,30 @@ defmodule Explorer.DataFrame do
       >
   """
   @doc type: :single
-  @spec group_by(df :: DataFrame.t(), groups_or_group :: [String.t()] | String.t()) ::
+  @spec group_by(df :: DataFrame.t(), groups_or_group :: column_names() | column_name()) ::
           DataFrame.t()
   def group_by(df, groups) when is_list(groups) do
-    names = names(df)
-    Enum.each(groups, fn name -> maybe_raise_column_not_found(names, name) end)
+    groups = to_existing_columns(df, groups)
+
+    all_groups =
+      df.groups
+      |> MapSet.new()
+      |> MapSet.union(MapSet.new(groups))
+      |> MapSet.to_list()
```

Comment:
I am almost sure that `Enum.uniq(df.groups ++ groups)` will be more efficient. :)